tv_apps/smart-system/js/app_install_manager.js

```diff
@@ -338,11 +338,30 @@ var AppInstallManager = {
       this.systemBanner.show({
         id: 'preview-app-hint'
       });
+      this.previewOpened[manifestURL] = ++this.previewOpened[manifestURL] || 1;
     };
 
     var handlePreviewTerminated = () => {
       window.removeEventListener('previewterminated', handlePreviewTerminated);
-      this.uninstallPreviewApp();
+      var askAddPrompt = this.previewOpened[manifestURL] === this.openedTimes;
+      if (askAddPrompt && !this.getAppAddedState(manifestURL)) {
+        var options = { 'manifest': app.manifest };
+        var TYPES = AppInstallDialogs.TYPES;
+        // Keep app info in case of it is destroyed by terminated event
+        var cloneApp = {
+          manifestURL: app.manifestURL,
+          manifest: app.manifest,
+          updateManifest: app.updateManifest
+        };
+        this.appInstallDialogs.show(TYPES.AddAppDialog, options).then(
+          this.handleAddApp.bind(this, cloneApp),
+          this.handleAddAppCancel.bind(this)
```

Comment:
I guess we don't really need the `err` message when cancelling. How about passing `this.uninstallPreviewApp.bind(this)` directly here?